src/core/BufferGeometry.js

```diff
@@ -822,7 +822,7 @@ BufferGeometry.prototype = Object.assign( Object.create( EventDispatcher.prototy
 
 			var attributeSize = attribute2.itemSize;
 
-			for ( var i = 0, j = attributeSize * offset; i < attributeArray2.length; i ++, j ++ ) {
+			for ( var i = 0, j = attributeSize * offset; i < attributeArray2.length && j < attributeArray1.length; i ++, j ++ ) {
 
 				attributeArray1[ j ] = attributeArray2[ i ];
```

Comment:
`j` can be bigger than the `attributeArray1.length` here. But this method seems to expect `attributeArray1` size won't change.

`attributeArray1/2` are typed array then `attributeArray1` size won't grow even if we set `attributeArray1[ attributeArray1.length ] = value;`. But I think better to clarify the loop end condition. It'll be efficient in case `j` is much bigger than `attributeArray1.length`.